src/scanner/patterns.rs

```diff
@@ -71,8 +86,30 @@ impl Patterns {
                     .ok_or(PatternsError::PathHasNoParent)?,
             )?;
 
+            let mut headers = header::HeaderMap::new();
+
+            if let Some(path) = &self.server.auth_token_file_path {
+                let auth_token =
+                    fs::read_to_string(path).map_err(PatternsError::CouldNotOpenAuthTokenFile)?;
```

Comment:
For curiosity why did you select a separate file for this? Was it for easier secrets management as the config is not particularly sensitive and you can keep them in separate places?

Reply:
Yeah it was so that tools that implement a login sub-command could just write to the file once they get a token rather than having to update the config. That being said, one additional thing I need to add is:

If `~/.config/leaktk/pattern-server-auth-token` exists and `auth_token_file_path` isn't defined, use that file by default.

Reply:
Updated so that the config if set and contains the actual auth token (instead of a path) it wins out but if it's not set you can fall back on LEAKTK_PATTERN_SERVER_AUTH_TOKEN then ~/.config/leaktk/pattern-server-auth-token.